Sources/Valkey/Connection/ValkeyConnection.swift

```diff
@@ -153,16 +175,44 @@ public final actor ValkeyConnection: ValkeyClientProtocol, Sendable {
 
     @inlinable
     func _execute<Command: ValkeyCommand>(command: Command) async throws -> RESPToken {
+        #if DistributedTracingSupport
+        let span = self.tracer?.startSpan(Command.name, ofKind: .client)
+        defer { span?.end() }
+
+        span?.updateAttributes { attributes in
+            self.applyCommonAttributes(to: &attributes, commandName: Command.name)
+        }
+        #endif
+
         let requestID = Self.requestIDGenerator.next()
-        return try await withTaskCancellationHandler {
-            if Task.isCancelled {
-                throw ValkeyClientError(.cancelled)
+
+        do {
+            return try await withTaskCancellationHandler {
+                if Task.isCancelled {
+                    throw ValkeyClientError(.cancelled)
+                }
+                return try await withCheckedThrowingContinuation { continuation in
+                    self.channelHandler.write(command: command, continuation: continuation, requestID: requestID)
+                }
+            } onCancel: {
+                self.cancel(requestID: requestID)
             }
-            return try await withCheckedThrowingContinuation { continuation in
-                self.channelHandler.write(command: command, continuation: continuation, requestID: requestID)
+        } catch let error as ValkeyClientError {
+            #if DistributedTracingSupport
+            if let span {
+                span.recordError(error)
+                span.setStatus(SpanStatus(code: .error))
+                if let prefix = error.simpleErrorPrefix {
+                    span.attributes["db.response.status_code"] = "\(prefix)"
+                }
             }
-        } onCancel: {
-            self.cancel(requestID: requestID)
+            #endif
+            throw error
+        } catch {
+            #if DistributedTracingSupport
+            span?.recordError(error)
```

Comment:
Is it intentional that this code path doesn't set the status like the one above?